infrastructure/tooling/src/smoketest/checks/role.js

```diff
@@ -28,12 +28,16 @@ exports.tasks.push({
     };
     assert.deepEqual(expandedRole.scopes, expectedScopes.scopes);
     const query = {};
-    const anHourAgo = Date.now() - (1000*60*60);
+    const date = new Date();
+    const currentHour = date.getHours();
     while (1) {
       const res = await auth.listRoles2();
-      for(let i=0;i<res.roles.length;i++){
-        if(res.roles[i].roleId.includes('project:taskcluster:smoketest:') && res.roles[i].lastModified < anHourAgo){
-          await auth.deleteRole(res.roles[i].roleId);
+      for(let index in res.roles){
+        if(res.roles[index].roleId.includes('project:taskcluster:smoketest:')){
+          const lastModified = (new Date(res.roles[index].lastModified)).getHours();
+          if(currentHour - lastModified >= 1){
+            await auth.deleteRole(res.roles[index].roleId);
+          }
```

Comment:
This only compares the hours, so for example something created at 23:00 two days ago would not be deleted if it's currently 7:00.

I think the issues in the previous version were around the types:

```
> res_roles_index_lastModified = '2019-10-29T23:23:31.391Z'
'2019-10-29T23:23:31.391Z'
> anHourAgo = Date.now() - (1000*60*60)
1572979864372
> res_roles_index_lastModified < anHourAgo
false
> new Date(res_roles_index_lastModified) < new Date(anHourAgo)
true
> 
```
so I think you could fix it by just calling `new Date(..)` with both the numeric value used to calculate `anHourAgo` and the string value we get in `res.roles[index].lastModified`.

Reply:
Yes. That is correct. Thank you.